tests/Sharpliner.Tests/AzureDevOps/TemplateTests.cs

```diff
@@ -46,37 +46,44 @@ public void TemplateList_Serialization_Test()
             """);
     }
 
-    private class Template_Definition : StepTemplateDefinition
+    private class Step_Template_Definition : StepTemplateDefinition
     {
         public override string TargetFile => "template.yml";
 
+        protected Parameter project = StringParameter("project");
+        protected Parameter version = StringParameter("version", allowedValues: [ "5.0.100", "5.0.102" ]);
+        protected Parameter skipBuild = BooleanParameter("skipBuild");
+        protected Parameter useNugetOrg = BooleanParameter("useNugetOrg", defaultValue: false);
+        protected Parameter restore = BooleanParameter("restore", defaultValue: true);
+        protected Parameter<Step> afterBuild = StepParameter("afterBuild", Bash.Inline("cp -R logs $(Build.ArtifactStagingDirectory)"));
+
         public override List<Parameter> Parameters =>
         [
-            StringParameter("project"),
-            StringParameter("version", allowedValues: [ "5.0.100", "5.0.102" ]),
-            BooleanParameter("skipBuild"),
-            BooleanParameter("useNugetOrg", defaultValue: false),
-            BooleanParameter("restore", defaultValue: true),
-            StepParameter("afterBuild", Bash.Inline("cp -R logs $(Build.ArtifactStagingDirectory)")),
+            project,
+            version,
+            skipBuild,
+            useNugetOrg,
+            restore,
+            afterBuild,
         ];
 
         public override ConditionedList<Step> Definition =>
         [
-            DotNet.Install.Sdk(parameters["version"]),
+            DotNet.Install.Sdk(version.Reference),
```

Comment:
in these cases what used to happen in that the `ParameterReference` type has an implicit operator to `string` so this worked.

I'm not sure about adding an implicit operator for string for the `Parameter` type will be clear, \
@premun  WDYT?

Reply:
I think it might just work because wherever you use this, you'd actually expect the `${{ parameter.name }}` result and if you would have this:
```
If.Equal(version, "8.0")
```

then it would do `${{ if eq(${{ parameter.version }}, "8.0" }} but I believe we already can recognize this and strip the inner `${{` away so it might just all work

Reply:
So maybe just `Parameter` -> `string` implicit conversion is all we need

Reply:
But I understand it's a dangerous thing that could happen invisibly in many places around your codebase and it can be dangerous

Reply:
I think the `.Reference` member is a good idea over all too, just to be clear. I am just thinking that maybe it won't be as discoverable and the auto-cast to string might just work out nice. After all, you don't handle the `Parameter` objects a lot and they do not get passed around much, so it might be just the right and nice thing to do.